spec/controllers/projects_controller_spec.rb

```diff
@@ -1,13 +1,85 @@
 require 'spec_helper'
 
+RSpec.shared_examples 'denies access' do
+  before { request }
+
+  it 'redirects to the root path' do
+    expect(response).to redirect_to(root_path)
+  end
+
+  it 'sets the flash' do
+    expect(flash[:alert]).to eq('You are not authorized to do that')
+  end
+end
+
+RSpec.shared_examples 'can update project' do
+  before do
+    allow(controller).to receive(:current_user).and_return(user)
+    allow(Project).to receive(:find).with(project.id).and_return(project)
+    allow(controller).to receive(:authorize!).with(:update, project)
+      .and_return(project)
+  end
+
+  context 'valid values' do
+    before do
+      allow(project).to receive(:update).and_return(true)
+    end
+
+    it 'calls the update method' do
+      expect(project).to receive(:update)
+      request
+    end
+
+    it 'returns a status of 200' do
+      expect(request).to have_http_status(200)
+    end
+  end
+
+  context 'invalid values' do
+    before do
+      allow(project).to receive(:update).and_return(false)
+      allow(project).to receive(:errors).and_return(errors)
+    end
+
+    it 'calls the update method' do
+      expect(project).to receive(:update)
+      request
+    end
+
+    it 'returns a status of 422' do
+      expect(request).to have_http_status(422)
+    end
+  end
+end
+
+RSpec.shared_examples 'can destroy project' do
```

Comment:
This shared example should just be in the tests for the destroy method, Its too coupled to that action to be a shared example.